presto-native-execution/presto_cpp/main/dynamic_registry/DynamicLibraryLoader.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "presto_cpp/main/dynamic_registry/DynamicLibraryLoader.h"
+
+#include <fstream>
+
+#include "presto_cpp/main/JsonSignatureParser.h"
+#include "presto_cpp/main/common/Configs.h"
+#include "velox/common/base/Exceptions.h"
+#include "velox/common/base/Fs.h"
+#include "velox/common/dynamic_registry/DynamicLibraryLoader.h"
+#include "velox/expression/SimpleFunctionRegistry.h"
+#include "velox/functions/FunctionRegistry.h"
+
+namespace facebook::presto {
+namespace {
+#ifndef DYNAMIC_LIB_FILE_EXT
+#error DYNAMIC_LIB_FILE_EXT must be defined in CMAKE
+#endif
+namespace fs = std::filesystem;
+
+bool isConfigSignatureRegistered(
+    const std::vector<const facebook::velox::exec::FunctionSignature*>&
+        registeredFnSignatures,
+    const facebook::velox::exec::FunctionSignaturePtr& localFnSignature) {
+  return std::find_if(
+             registeredFnSignatures.begin(),
+             registeredFnSignatures.end(),
+             [&](const facebook::velox::exec::FunctionSignature* sig) {
+               return *sig == *localFnSignature;
+             }) != registeredFnSignatures.end();
+}
+} // namespace
+
+std::string DynamicLibraryLoader::constructAbsoluteFilePath(
```

Comment:
It would be better for this to return an std::optional<std::string>, instead of returning empty string for an invalid path. Lets change it.